### DISCUSSION TITLE
AJAX requests to 3rd party custom fields return full HTML

### DISCUSSION THREAD

**Discussion by nikans** [laravel/nova-issues#3310] [Q&A] [answered]
- Laravel Version: 8.34.0
- Nova Version: 3.22.0
- PHP Version: 7.4.16
- MacOS, Docker, Postgres
- Local env

### Description:

I've tried several 3rd party custom fields for many-to-many management, while viewing and deleting relations work, XHR requests to their respective URLs always return full HTML document instead of data. I'm starting to think, something is wrong with my system, but don't know where to look. 

### Detailed steps to reproduce the issue on a fresh Nova installation:

A basic working many-to-many relation with pivot table. Nova's `BelongsToMany` field works as expected. 

Tested with these packages:
- https://novapackages.com/packages/ziffmedia/nova-select-plus
- https://novapackages.com/packages/dillingham/nova-attach-many
- https://novapackages.com/packages/benjacho/belongs-to-many-field

<img width="500" alt="Screenshot 2021-03-25 at 08 58 13" src="https://user-images.githubusercontent.com/1039100/112426035-42a1e500-8d48-11eb-8996-2b499d09e952.png">

Browser console shows full HTML document instead of data.

<img width="500" alt="Screenshot 2021-03-25 at 08 59 08" src="https://user-images.githubusercontent.com/1039100/112426103-5fd6b380-8d48-11eb-857d-ff94ccc73f7a.png">

Logging from custom fields controllers doesn't help — looks like the request never gets there (works from fields' boot).

<img width="400" alt="Screenshot 2021-03-25 at 09 00 45" src="https://user-images.githubusercontent.com/1039100/112426215-99a7ba00-8d48-11eb-9dac-0f4be6fd8b2d.png">

Guidance is much appreciated.

**Comment by crynobone** [comment]
Laravel Nova doesn't have any controller with `options()` method. I would recommend creating an issue on the respective field repositories:

* https://github.com/ziffmedia/nova-select-plus
* https://github.com/dillingham/nova-attach-many
* https://github.com/Benjacho/belongs-to-many-field-nova

**Comment by nikans** [comment, marked as answer]
Yeah, I know it doesn't, @crynobone.

These three custom fields use different controllers for requests, but non of them are actually reached — and all of them return  plain HTML instead of data. It looks like something in my Nova is broken, not in those packages. Don't know where to look, though.

**Reply by crynobone** [reply]
I would assume all the requests should go to respective routes, such as:

https://github.com/dillingham/nova-attach-many/blob/1df678fe937f113d15c8de475e458cf04e89af67/src/Providers/FieldServiceProvider.php#L22-L26

From there you can use `route:list` to check whether the routes are still available. XHR request returning HTML seem to indicate routes has been overriden.

> p/s: moved this to Discussion as this not a bug report, but rather Q&A/Support

**Reply by nikans** [reply]
Thanks for the direction.

I had to fix a routing error, of which I found out only by calling `route:list` (Laravel 8.34.0 is being broken by Routes with a closure, though looks like Nova/Laravel pre-package one in `app/routes/api.php`). 

After that, `php artisan route:cache`.
